source/words.d.ts

```diff
@@ -1,11 +1,46 @@
-import type {IsLowerCase, IsNumeric, IsUpperCase, WordSeparators} from './internal';
+import type {
+	IsLowerCase,
+	IsNumeric,
+	IsUpperCase,
+	WordSeparators,
+} from './internal';
 
 type SkipEmptyWord<Word extends string> = Word extends '' ? [] : [Word];
 
-type RemoveLastCharacter<Sentence extends string, Character extends string> = Sentence extends `${infer LeftSide}${Character}`
+type RemoveLastCharacter<
+	Sentence extends string,
+	Character extends string,
+> = Sentence extends `${infer LeftSide}${Character}`
 	? SkipEmptyWord<LeftSide>
 	: never;
 
+/**
+Words options.
+
+@see {@link Words}
+*/
+export type WordsOptions = {
+	/**
+	Split on numeric sequence.
+
+	@default true
+
+	@example
+	```
+	type Example1 = Words<'p2pNetwork', {splitOnNumbers: true}>;
+	//=> ["p", "2", "p", "Network"]
+
+	type Example2 = Words<'p2pNetwork', {splitOnNumbers: false}>;
+	//=> ["p2p", "Network"]
+	```
+	*/
+	splitOnNumbers?: boolean;
+};
+
+type DefaultOptions = {
+	splitOnNumbers: true;
```

Comment:
Should this default to `false`? 4.36.0 seems to be a breaking change since I have to set this to false to avoid errors using `KebabCase`.